standalone/pruntime/enclave/src/msg/msg_tunnel.rs

```diff
@@ -0,0 +1,133 @@
+use std::cell::RefCell;
+use std::borrow::BorrowMut;
+use crate::std::vec::Vec;
+use crate::std::sync::SgxMutex;
+
+
+lazy_static! {
+    pub static ref  STATIC_MSG_TUNNEL_MUT:SgxMutex<MsgTunnel> = {
+        let  m:MsgTunnel = MsgTunnel::default();
+        SgxMutex::new(m)
+    };
+}
+
+
+#[derive(Debug, Clone)]
+pub struct MsgTunnel {
+    list: RefCell<Vec<MsgTypeWrapper>>
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum MsgType {
+    BALANCE,
+    WORKER,
+}
+
+#[derive(Debug, Clone)]
+pub struct MsgTypeWrapper {
+    pub msg_type: MsgType,
+    pub msg_data: MsgData,
+}
+
+
+#[derive(Debug, Clone)]
+pub struct MsgData {
+    pub sequence: RefCell<u64>,
+    pub queue: RefCell<Vec<RefCell<MsgDetail>>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MsgDetail {
+    pub msg_id: u64,
```

Comment:
`seq_id` works as `msg_id` in my understanding. We prefer to make the design as compact as possible, not redundancy. Also I think it's better to embed the actual message body into the struct by a generic type, rather than putting a `Vec<u8>`, as mentioned in the long comment above